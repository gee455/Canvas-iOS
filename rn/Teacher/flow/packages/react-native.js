//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
