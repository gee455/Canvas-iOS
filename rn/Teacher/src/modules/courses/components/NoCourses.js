//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
} from 'react-native'
import i18n from 'format-message'
import { Text } from '../../../common/text'
import { Button } from '../../../common/buttons'
import colors from '../../../common/colors'

type Props = {
  onAddCoursePressed: () => void,
  style?: any,
}

export class NoCourses extends Component<Props> {
  props: Props

  render () {
    let welcome = i18n('Welcome!')

    let bodyText = i18n('Add a few of your favorite courses to make this place your home.')

    let buttonText = i18n('Add Courses')

    return (
      <View style={[styles.container, this.props.style]}>
        <Text style={styles.welcome} testID='no-courses.welcome-lbl'>{welcome}</Text>
        <Text
          style={styles.paragraph} testID='no-courses.description-lbl'>{bodyText}</Text>
        <Button
          accessibilityLabel={buttonText}
          testID='no-courses.add-courses-btn'
          onPress={this.props.onAddCoursePressed}
          style={styles.button}
          containerStyle={styles.buttonContainer}
        >
          {buttonText}
        </Button>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 40,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  welcome: {
    fontSize: 30,
    lineHeight: 36,
    color: colors.darkText,
    marginBottom: 8,
  },
  paragraph: {
    textAlign: 'center',
    color: colors.grey4,
    fontSize: 16,
    lineHeight: 19,
    paddingHorizontal: 10,
    marginBottom: 40,
  },
  button: {
    fontSize: 16,
    lineHeight: 19,
  },
  buttonContainer: {
    borderRadius: 4,
    paddingHorizontal: 24,
    paddingVertical: 16,
  },
})

export default (NoCourses: any)
