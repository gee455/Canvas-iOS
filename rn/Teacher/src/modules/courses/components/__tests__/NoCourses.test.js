//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import 'react-native'
import React from 'react'
import { NoCourses } from '../NoCourses'
import explore from '../../../../../test/helpers/explore'

jest.mock('TouchableOpacity', () => 'TouchableOpacity')

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

let defaultProps = {
  onAddCoursePressed: () => {},
}

test('renders NoCourses correctly', () => {
  let tree = renderer.create(
    <NoCourses {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('calls onAddCoursePressed when button is pressed', () => {
  let func = jest.fn()
  let tree = renderer.create(
    <NoCourses {...defaultProps} onAddCoursePressed={func} />
  ).toJSON()

  let button: any = explore(tree).selectByID('no-courses.add-courses-btn')
  button.props.onPress()
  expect(func).toHaveBeenCalled()
})
