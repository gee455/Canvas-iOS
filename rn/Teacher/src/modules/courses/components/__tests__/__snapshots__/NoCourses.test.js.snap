// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NoCourses correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingHorizontal": 40,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 30,
          "lineHeight": 36,
          "marginBottom": 8,
        },
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
    testID="no-courses.welcome-lbl"
  >
    Welcome!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Object {
          "color": "#8B969E",
          "fontSize": 16,
          "lineHeight": 19,
          "marginBottom": 40,
          "paddingHorizontal": 10,
          "textAlign": "center",
        },
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
    testID="no-courses.description-lbl"
  >
    Add a few of your favorite courses to make this place your home.
  </Text>
  <TouchableOpacity
    accessibilityComponentType="button"
    accessibilityTraits="button"
    activeOpacity={0.2}
    delayLongPress={undefined}
    delayPressIn={undefined}
    delayPressOut={undefined}
    onLongPress={undefined}
    onPress={[Function]}
    onPressIn={undefined}
    onPressOut={undefined}
    style={
      Array [
        Object {
          "backgroundColor": "#374A59",
          "borderRadius": 8,
          "overflow": "hidden",
          "padding": 10,
        },
        Object {
          "backgroundColor": "#374A59",
        },
        Object {
          "borderRadius": 4,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
    testID="no-courses.add-courses-btn"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          Array [
            Object {
              "color": "white",
            },
            Object {
              "color": "white",
            },
            Object {
              "fontSize": 16,
              "lineHeight": 19,
            },
          ],
          null,
        ]
      }
    >
      Add Courses
    </Text>
  </TouchableOpacity>
</View>
`;
